lib/components/ToggleSwitch/index.tsx

```diff
@@ -0,0 +1,14 @@
+import React, { ChangeEvent, forwardRef } from 'react';
+import './ToggleSwitch.scss';
+
+interface ToggleSwitchProps {
+    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+    value: boolean;
+}
+
+export const ToggleSwitch = forwardRef<HTMLInputElement, ToggleSwitchProps>(({ onChange, value }, ref) => (
+    <label className='deriv-toggle-switch'>
+        <input checked={value} onChange={onChange} ref={ref} type='checkbox' />
+        <span className='deriv-toggle-switch__slider' />
+    </label>
+));
```

Comment:
the reason is that the component itself should be able to internally handle the very basic behaviour which is toggling the value and if the user wants to add more stuff to it then they can pass onChange as well. we did the same thing for Tabs component as well.
what do you think @yaswanth-deriv 

```suggestion
export const ToggleSwitch = forwardRef<HTMLInputElement, ToggleSwitchProps>(({ onChange, value }, ref) => {
    const [checked, setIsChecked] = React.useState(value || false);
    const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
        setIsChecked(!checked);
        onChange(event);
    }
 return (
    <label className='deriv-toggle-switch'>
        <input checked={value} onChange={handleOnChange} ref={ref} type='checkbox' />
        <span className='deriv-toggle-switch__slider' />
    </label>
  )
));
```